### PR TITLE
Feat: Woodcutting

### PR DESCRIPTION
**Pull request by rooooose-b**
From the plugin "woodcutting", but fixed to include stripped logs and recipes for pressure plates.

### CODE REVIEW

File: src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            8,
+            4,
+            4,
+            4,
+            4,
+            8,
+            4,
+            4,
+            1
+        };
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java:43]:
```suggestion		
        Map<String, Integer> map = Map.of(
			"BUTTON", 4,
			"FENCE", 8,
			"FENCE_GATE", 4,
			"PLANKS", 4,
			"PRESSURE_PLATE", 4,
			"SIGN", 4,
			"SLAB", 8,
			"STAIRS", 4,
			"TRAPDOOR", 4,
			"DOOR", 1
		);
```
This is less verbose and error-prone if someone were to modify this in the future.

File: src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        for (Material log : Tag.LOGS.getValues()) {
+            String logString = log.toString();
+            if (logString.contains("STRIPPED")) {
+                logString = logString.substring(9);
+            }
+            String[] materialArray = logString.split("_");
+
+            if (materialArray.length > 2 && materialArray[2] != null) {
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java:52]: At a quick glance, I'm assuming that this handles stripped logs. I think a comment would be good to note what the purpose and logic of the clause is.

File: src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+            for (int i = 0; i < woodcuttingItems.length; i++) {
+                String item = woodcuttingItems[i];
+                int count = woodcuttingCount[i];
+
+                Material itemName = Material.getMaterial(materialArray[0].concat("_").concat(item));
+
+                ItemStack itemStack = new ItemStack(itemName, count);
+                itemStackList.add(itemStack);
+            }
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java:66]:
```suggestion
            map.forEach((item, count) -> {
                Material itemName = Material.getMaterial(materialArray[0].concat("_").concat(item));

                ItemStack itemStack = new ItemStack(itemName, count);
                itemStackList.add(itemStack);
            });
```

File: src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+            String key = log.name() + Math.random();
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java:68]: Pretty sure namespacedkeys shouldn't be randomized. @NuclearDonut47 thoughts?

**Review by darksaid98 — CHANGES_REQUESTED** [review]
Good work, I'm being intentionally nitpicky here with changes in order to reduce further clutter in this repo.

File: src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java (hunk 2 above)
Comment (rooooose-b) [reply src/main/java/me/ShermansWorld/AlathraExtras/woodcutting/Woodcutting.java:39]: The if-statement right above this section handles stripped logs. This if-statment (line 52) covers dark oak. I'll add comments on both.

**Review by darksaid98 — APPROVED** [review]

**Review by NuclearDonut47 — APPROVED** [review]
Made adjustments. Base functionality is preserved from prior commits. Null check safety and minor cleanup added. Ready for merge.

**Pull request merged**